### DISCUSSION TITLE
Consuming a valtio proxy from another package results in "TypeError: proxyState is not iterable"

### DISCUSSION THREAD

**Discussion by iam4x** [pmndrs/valtio#706] [Q&A] [answered]
## Summary

- npm module expose a valtio proxy:

```tsx
import {createExchange} from 'foo';

const exchange = createExchange();
// exchange.store is a Proxy object from valtio/vanilla

const Component = () => {
  const snap = useSnapshot(exchange.store);
  return (<div>{snap.foo}</div>);
}
```

This doesn't work and throws:

```
TypeError: proxyState is not iterable
```

I've logged: `console.log(Object.getOwnPropertyDescriptors(exchange.store))` which outputs:

![image](https://user-images.githubusercontent.com/893837/229754574-16781bed-ac4f-49d2-91b4-4c6125dad589.png)

* I've tried using `proxy()` from `valtio` instead of `valtio/vanilla`
* I've tried npm link the external package
* I'm sure application & package are using same valtio version

Do you have an idea?

**Comment by iam4x** [comment]
Still debugging, it looks like its importing .mjs and .js packages:

![image](https://user-images.githubusercontent.com/893837/229761482-73665deb-cb61-4481-96e0-3dd9a3f32b2a.png)


So `proxyStateMap` is correctly set in one context but not the other

**Comment by dai-shi** [comment, marked as answer]
It's called dual package hazard. (which is not supported in valtio.)
You have to use only one package.
In the future version of valtio, we may drop `.js` modules.

**Comment by iam4x** [comment]
Yeah ok, thank I'll just fork and release another version under my scope, using one package is not an option.
Do you have any guidance? I guess I have just to comment into `rollup.config.js` the builds I don't want.

**Reply by dai-shi** [reply]
If you were to fork valtio, defining shared state on `globalThis` might help.

**Reply by iam4x** [reply]
Thanks, all good now.
If anyone runs into the same issue: https://www.npmjs.com/package/@iam4x/valtio

**Comment by capaj** [comment]
I got this error when I use valtio in a pnpm monorepo. running `pnpm dedupe` fixes this problem.
